test/bats/test.bats

```diff
@@ -0,0 +1,58 @@
+#!/usr/bin/env bats
+
+load helpers
+
+BATS_TESTS_DIR=test/bats/tests
+WAIT_TIME=60
+SLEEP_TIME=1
+IMAGE_TAG=e2e-$(git rev-parse --short HEAD)
+
+setup() {
+  if [[ -z "${AZURE_CLIENT_ID}" ]] || [[ -z "${AZURE_CLIENT_SECRET}" ]]; then
+    echo "Error: Azure service principal is not provided" >&2
+    return 1
+  fi
+}
+
+@test "install helm chart with e2e image" {
+  run helm install charts/secrets-store-csi-driver -n csi-secrets-store --namespace dev \
+          --set provider="" \
+          --set image.pullPolicy="IfNotPresent" \
+          --set image.repository="e2e/secrets-store-csi" \
+          --set image.tag=$IMAGE_TAG
+  assert_success
+}
+
+@test "csi-secrets-store-attacher-0 is running" {
+  cmd="kubectl wait -n dev --for=condition=Ready --timeout=60s pod/csi-secrets-store-attacher-0"
+  wait_for_process $WAIT_TIME $SLEEP_TIME "$cmd"
+
+  run kubectl get pod/csi-secrets-store-attacher-0 -n dev
+  assert_success
+}
+
```

Comment:
nit: would be cool to also add a test for `secretproviderclasses` in this PR. or we can add it as a followup PR if you want.

we should test for creation of https://github.com/deislabs/secrets-store-csi-driver/blob/master/charts/secrets-store-csi-driver/templates/secrets-store.csi.k8s.com_secretproviderclasses.yaml
`kubectl get crd/secretproviderclasses.secrets-store.csi.k8s.com`

and 

test for creation the `azure` secretproviderclasses similar to https://github.com/deislabs/secrets-store-csi-driver/blob/master/secretProviderClass/config/samples/v1alpha1_secretproviderclass.yaml
`kubectl get secretproviderclasses`


then test for a pod using `secretProviderClass: "azure"` https://github.com/deislabs/secrets-store-csi-driver/blob/master/pkg/providers/azure/examples/nginx-pod-secrets-store-inline-volume-crd.yaml